internal/services/inference/deployment.go

```diff
@@ -0,0 +1,308 @@
+							Optional:    true,
+						},
+						"disable_auth": {
+							Type:        schema.TypeBool,
+							Description: "Disable the authentication on the endpoint.",
+							Optional:    true,
+							Default:     false,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ResourceDeploymentCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, err := NewAPIWithRegion(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	req := &inference.CreateDeploymentRequest{
+		Region:    region,
+		ProjectID: d.Get("project_id").(string),
+		Name:      d.Get("name").(string),
+		NodeType:  d.Get("node_type").(string),
+		ModelName: d.Get("model_name").(string),
+		Tags:      types.ExpandStrings(d.Get("tags")),
+	}
+
+	endpoint := inference.EndpointSpec{
+		Public:         nil,
+		PrivateNetwork: nil,
+		DisableAuth:    false,
+	}
+
+	if _, isEndpoint := d.GetOk("endpoints"); isEndpoint {
+		if publicEndpoint := d.Get("endpoints.0.public_endpoint"); publicEndpoint != nil && publicEndpoint.(bool) {
+			endpoint.Public = &inference.EndpointSpecPublic{}
+		}
+		if privateEndpoint := d.Get("endpoints.0.private_endpoint"); privateEndpoint != "" {
+			endpoint.PrivateNetwork = &inference.EndpointSpecPrivateNetwork{
+				PrivateNetworkID: regional.ExpandID(privateEndpoint.(string)).ID,
+			}
+		}
+	}
+
+	req.Endpoints = []*inference.EndpointSpec{&endpoint}
+
+	if isAcceptingEula, ok := d.GetOk("accept_eula"); ok {
+		req.AcceptEula = scw.BoolPtr(isAcceptingEula.(bool))
+	}
+
+	deployment, err := api.CreateDeployment(req, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(regional.NewIDString(region, deployment.ID))
+
+	_, err = waitForDeployment(ctx, api, region, deployment.ID, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	return ResourceDeploymentRead(ctx, d, m)
+}
+
+func ResourceDeploymentRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, id, err := NewAPIWithRegionAndID(m, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	deployment, err := waitForDeployment(ctx, api, region, id, d.Timeout(schema.TimeoutRead))
+	if err != nil {
+		if httperrors.Is404(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+
+	_ = d.Set("name", deployment.Name)
+	_ = d.Set("region", deployment.Region)
+	_ = d.Set("project_id", deployment.ProjectID)
+	_ = d.Set("node_type", deployment.NodeType)
+	_ = d.Set("model_name", deployment.ModelName)
+	_ = d.Set("min_size", int(deployment.MinSize))
+	_ = d.Set("max_size", int(deployment.MaxSize))
+	_ = d.Set("size", int(deployment.Size))
+	_ = d.Set("status", deployment.Status)
+	_ = d.Set("model_id", deployment.ModelID)
+	_ = d.Set("created_at", types.FlattenTime(deployment.CreatedAt))
+	_ = d.Set("updated_at", types.FlattenTime(deployment.UpdatedAt))
+
+	for _, endpoint := range deployment.Endpoints {
```

Comment:
Should this export as `endpoints` ? User configure its `endpoints` but get different fields after StateRead.

Reply:
It doesn't make much sense to have a field at the root of the ressource being updated with a sub resource downstream :/ would it be better to have all information about endpoint at the same place?